src/signals/index.ts

```diff
@@ -1,3 +1,24 @@
-export function placeholder() {
-	return 'placeholder';
-}
\ No newline at end of file
+import { PartialStateUpdater, patchState, StateSignal } from '@ngrx/signals';
+import { immerReducer } from 'ngrx-immer';
+
+export type ImmerStateUpdater<State extends object> = (state: State) => void;
+
+function toFullStateUpdater<State extends object>(updater: PartialStateUpdater<State & {}> | ImmerStateUpdater<State & {}>): (state: State) => State | void {
+	return (state: State) => {
+		const patchedState = updater(state);
+		if (patchedState) {
+			return ({ ...state, ...patchedState });
+		}
+		return;
+	};
+}
+
+export function immerPatchState<State extends object>(state: StateSignal<State>, ...updaters: Array<Partial<State & {}> | PartialStateUpdater<State & {}> | ImmerStateUpdater<State & {}>>) {
```

Comment:
I wonder if we can update the generics to be able to do the following:

- the second patch updates the state and knows there a request state
- the first patch does the same but the state gets narrowed down because of the use of `setAllEntities`

![image](https://github.com/timdeschryver/ngrx-immer/assets/28659384/43f1f235-5e4f-4d6d-95ca-4b300c1de02c)

Reply:
Something like the following seems to work:

```
export declare function immerPatchState<State extends object, T = State>(state: StateSignal<T>, ...updaters: Array<Partial<T & {}> | PartialStateUpdater<T & {}> | ImmerStateUpdater<T & {}>>): void;
```

![image](https://github.com/timdeschryver/ngrx-immer/assets/28659384/63139939-cd91-4da4-b7fb-558be900fcde)

Reply:
Hi Tim, I can't currently work on it, but please update my branch as you wish. I'll continue tomorrow.